lib/kernel-manager.coffee

```diff
@@ -11,6 +11,7 @@ module.exports =
 class KernelManager
     constructor: ->
         @_runningKernels = {}
+        @_kernelSpecs = []
```

Comment:
On 11/07/16 12:59, Lukas Geiger wrote:
> In lib/kernel-manager.coffee <https://github.com/nteract/hydrogen/pull/337#discussion_r70244433>:
> 
>> @@ -11,6 +11,7 @@ module.exports =
>>  class KernelManager
>>      constructor: ->
>>          @_runningKernels = {}
>> +        @_kernelSpecs = []
> 
> I don't see why initialising |@_kernelSpecs = @parseKernelSpecSettings()| would be an improvement. Since on activation of Hydrogen |@updateKernelSpecs| is called anyway.

It's an improvement for users like me that cannot run `ipython kerlnelspec`.

It means I can start Hydrogen without waiting for `jupyter kernelspec` and `ipython kernelspec` to fail (which is the current behaviour).

> 
>     One thing doesn't preclude the other, but it makes me think that |@updateKernelSpecs| could be split into two methods |@getKernelSpecsFromSettings| and |@getKernelSpecsFromJupyter|.
> 
> This is actually the case right now. The only difference are the method names:
> |@updateKernelSpecs| is now called |@saveKernelSpecs|,
> |@getKernelSpecsFromJupyter| is |@updateKernelSpecs| and
> |@getKernelSpecsFromSettings| is |@parseKernelSpecSettings|.
> 
> We can rename those methods if you think this would make it clearer.

It makes sense to me. How do you think?